src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.example.onlinebookstore.repository;
+
+import com.example.onlinebookstore.exception.DataProcessingException;
+import com.example.onlinebookstore.model.Book;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.getTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Can`t save book"
+                    + book.getTitle()
+                    + " to DB");
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return book;
```

Comment:
It's time to learn such magic as `inTransaction()` method :)
```suggestion
        try {
            sessionFactory.inTransaction(session -> session.persist(book));
            return book;
        } catch (Exception e) {
            throw new YourExceptionHere("Cannot save new book", e);
        }
```